engine/classes/Elgg/Search/SearchService.php

```diff
@@ -234,7 +234,7 @@ public function normalizeSearchFields(array $options = []) {
 		} else {
 			// only allow known fields
 			foreach ($fields as $property_type => $property_type_fields) {
-				if (empty($options['fields'][$property_type])) {
+				if (empty($property_type_fields)) {
```

Comment:
I don't think that's what's intended here. Let's see, but tests will likely fail

Reply:
i believe it is... you will want to reset the allowed fields if non are allowed/known

Reply:
That's right, but it should happen before this loop. This loops normalizes the known fields.

Reply:
tests are failing as expected, but maybe the test assume the wrong thing...

this foreach should do the following if i am correct:
if provided `$options['fields'][$type]` contains fields that are not present in the allowed `$fields` they should be removed... thus if `$fields[$type]`  is empty `$options['fields'][$type]` should also be empty...

Reply:
There is difference between empty and not set. If the fields are not set, we default to known fields, if they are set as false or [], then we don't use any.